### PR TITLE
(💄✅) : New field for activities and UI fixes for activity/children pages 

### PR DESCRIPTION
**Pull request by d-azmanov**
- Added a boundary field for activities and updated the database records to include this field.
- Added a limit on the number of children a parent can add.
- Mobile styling enhancements for the following pages:

> Add activity

> Edit activity

> View activity

> Add child

> Edit child

### CODE REVIEW

**Review by PxttytheThxtty — DISMISSED** [review]
Nice

File: libs/add-child/feature/src/lib/add-child.component.spec.ts [hunk 1]
```diff
@@ -62,6 +62,9 @@ describe('AddChildComponent', () => {
       aupair: ''
     };
 
+    component.allChildren = [expectedValue];
+    
+
```
Comment (ricecakesreece) [inline libs/add-child/feature/src/lib/add-child.component.spec.ts:67]: Extra line

File: libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss [hunk 2]
```diff
@@ -47,9 +47,10 @@ h2{
     text-align: right;
 }
 
-// Button styling
+// Form input styling
 .activity-details input{
-    width: 411px;
+    // max-width: 411px;
```
Comment (ricecakesreece) [inline libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss:52]: Code left in

File: libs/parent-add-activity/feature/src/lib/parent-add-activity.component.ts [hunk 3]
```diff
@@ -213,10 +268,13 @@ export class ParentAddActivityComponent implements OnInit{
...
         return;
       }
+
+      //Clear previous suggested locations
+      this.potentialLocations.splice(0);
   
       //Add returned data to the array
       const len = res.length;
-      for (let j = 0; j < len && j<4; j++) 
+      for (let j = 0; j < len && j<10; j++)
```
Comment (ricecakesreece) [inline libs/parent-add-activity/feature/src/lib/parent-add-activity.component.ts:277]: Why was this value changed?

File: libs/parent-edit-activity/feature/src/lib/parent-edit-activity.component.scss [hunk 4]
```diff
@@ -191,3 +283,41 @@ textarea:focus{
...
+@media (max-width: 411px) {
+    .activity-wrapper{
+        height: fit-content;
+        min-width: 290px;
+        margin-top: 60px;
+        margin-left: auto;
+        margin-right: auto;
+        background-color: transparent;
+        border-radius: 16px;
+        // padding-bottom: 7%;
```
Comment (ricecakesreece) [inline libs/parent-edit-activity/feature/src/lib/parent-edit-activity.component.scss:296]: Commented out code

File: libs/parent-edit-activity/feature/src/lib/parent-edit-activity.component.scss [hunk 5]
```diff
@@ -161,6 +161,98 @@ textarea:focus{
     color: red;
 }
 
+//Label styling
+//Labelling styles
```
Comment (ricecakesreece) [inline libs/parent-edit-activity/feature/src/lib/parent-edit-activity.component.scss:165]: Same comment

File: libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ion-content{
-    --ion-background-color:var(--ion-color-medium);
+    --ion-background-color:#EFEFEF !important;
```
Comment (ricecakesreece) [inline libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss:2]: --ion-background-color

File: libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss [hunk 7]
```diff
@@ -54,14 +56,13 @@ h2{
...
-    text-align: right;
 }
 
-// Button styling
+// Form input stylings
 .activity-details input{
     width: 411px;
     height: 62px;
-    background: var(--ion-background-color);
+    background: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss:65]: --ion-color-light-white

File: libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss [hunk 8]
```diff
@@ -72,7 +73,7 @@ h2{
     resize: none;
     width: 411px;
     height: 169px;
-    background: var(--ion-background-color);
+    background: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss:76]: --ion-color-light-white

File: libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss [hunk 9]
```diff
@@ -185,18 +278,42 @@ textarea:focus{
...
-        width: 100%;
+        height: fit-content;
+        min-width: 0px;
         margin-top: 60px;
         margin-left: auto;
         margin-right: auto;
-        background-color: transparent !important;
+        background-color: transparent;
         border-radius: 16px;
+        // padding-bottom: 7%;
```
Comment (ricecakesreece) [inline libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss:291]: Commented out code

File: libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss [hunk 10]
```diff
@@ -47,9 +47,10 @@ h2{
...
+    // min-width: 411px;
     height: 62px;
     background: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss:55]: --ion-color-light-white

Was added as part of theme

File: libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss [hunk 11]
```diff
@@ -71,7 +72,8 @@ label{
 }
 
 input[type=text], select{
-    width: 411px;
+    max-width: 411px !important;
+    min-width: 411px !important;
     height: 62px;
     background: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss:78]: --ion-color-light-white

File: libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss [hunk 12]
```diff
@@ -93,7 +95,19 @@ input:focus{
...
+.buttons{
+    display: grid;
+    grid-template-columns: repeat(2, 1fr);
+    gap: 15px;
+    width: 411px;
+    height: auto;
+    text-align: center;
+    margin-bottom: 100px;
+}
+
```
Comment (ricecakesreece) [inline libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss:107]: Space

File: libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss [hunk 13]
```diff
@@ -103,6 +117,57 @@ input:focus{
...
+    width: 195px !important;
+    height: 60px;
+    left: 973px;
+    top: 980px;
+    font-style: normal;
+    font-weight: 700;
+    font-size: 16px;
+    line-height: 19px;
+    color: #69A88D;
```
Comment (ricecakesreece) [inline libs/parent-add-activity/feature/src/lib/parent-add-activity.component.scss:136]: --ion-color-primary

File: libs/parent-edit-activity/feature/src/lib/parent-edit-activity.component.scss [hunk 14]
```diff
@@ -161,6 +161,98 @@ textarea:focus{
...
+
+#childLabel{
+    position: inherit;
+    text-align: left;
+    align-self: left;
+    margin-left: 5px;
+    margin-bottom: -30px;
+    margin-top: 30px;
+}
+
```
Comment (ricecakesreece) [inline libs/parent-edit-activity/feature/src/lib/parent-edit-activity.component.scss:254]: Spaces

File: libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss [hunk 15]
```diff
@@ -185,18 +278,42 @@ textarea:focus{
...
+        width: 95vw;
+    }
+
+    .buttons{
+        width: 95vw;
+        grid-template-columns: repeat(1, 1fr);
+    }
+
+    .cancel{
+        color: #69A88D;
```
Comment (ricecakesreece) [inline libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss:307]: --ion-color-primary

File: libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss [hunk 16]
```diff
@@ -185,18 +278,42 @@ textarea:focus{
...
+        background-color: transparent !important;
+        border: 2px solid #69A88D !important;
```
Comment (ricecakesreece) [inline libs/parent-view-activity/feature/src/lib/parent-view-activity.component.scss:309]: --ion-color-primary

**Review by PxttytheThxtty — APPROVED** [review]

**Review by ricecakesreece — APPROVED** [review]

**Pull request merged**